### PR TITLE
[IMP] project_todo: review the form view in To-Do

### PR DESCRIPTION
**Pull request by desh-odoo**
Before this commit, the form view displayed in To-Dos app are not
standard and a lot of custom is added to have a view like knowledge
article. The problem is that view is difficult to maintain without
adding more custom code.

This commit removes the custom code added for the form view and review
that form view to have a standard form view as we got in the other apps.
However, the only custom code added is to be able to hide the chatter by
default to have a full screen for the form view. The user will be able
to toggle the chatter as he wishes inside the form view.

task-3995765

### CODE REVIEW

File: addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/** @odoo-module */
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js:1]: don't need to add /** @odoo-module **/ to new JS Components 
```suggestion

```

File: addons/project_todo/static/src/components/todo_tag_ids/todo_tag_ids.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+
+/** @odoo-module */
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/components/todo_tag_ids/todo_tag_ids.js:2]:
```suggestion

/** @odoo-module */
```

same here

File: addons/project_todo/views/project_task_views.xml [hunk 3]
```diff
@@ -104,33 +104,26 @@
...
+                            options="{'color_field': 'color', 'no_create_edit': True}"/>
+                    <field name="user_ids"
+                            required="1"
+                            widget="many2many_avatar_user"/>
                     <field name="personal_stage_type_id"
                            domain="[('user_id','=',uid)]"
                            widget="statusbar"
                            options="{'clickable': '1', 'fold_field': 'fold'}"/>
-                    <field name="active" invisible="1"/>
-                    <field name="state" invisible="1"/>
```
Comment (ppr-odoo) [inline addons/project_todo/views/project_task_views.xml:127]: you can use fieldDependencies instead.

File: addons/project_todo/static/tests/todo_form_view.test.js [hunk 4]
```diff
@@ -27,6 +27,10 @@ beforeEach(() => {
         form: `
             <form string="To-do" class="o_todo_form_view" js_class="todo_form">
                 <field name="name"/>
+                <field name="tag_ids" invisible="1"/>
+                <field name="user_ids" invisible="1"/>
+                <field name="priority" invisible="1"/>
+                <field name="personal_stage_type_id" invisible="1"/>
```
Comment (ppr-odoo) [inline addons/project_todo/static/tests/todo_form_view.test.js:33]:
```suggestion
```
no needed

File: addons/project_todo/static/tests/todo_editable_breadcrumb_name.test.js [hunk 5]
```diff
@@ -30,6 +30,10 @@ beforeEach(() => {
             <form string="To-do" class="o_todo_form_view" js_class="todo_form">
                 <field name="name"/>
                 <field name="state" invisible="1"/>
+                <field name="tag_ids" invisible="1"/>
+                <field name="user_ids" invisible="1"/>
+                <field name="priority" invisible="1"/>
+                <field name="personal_stage_type_id" invisible="1"/>
```
Comment (ppr-odoo) [inline addons/project_todo/static/tests/todo_editable_breadcrumb_name.test.js:36]:
```suggestion
                <field name="tag_ids" invisible="1"/>
                <field name="user_ids" invisible="1"/>
                <field name="priority" invisible="1"/>
                <field name="personal_stage_type_id" invisible="1"/>
```

same here

File: addons/project_todo/static/tests/todo_done_checkmark.test.js [hunk 6]
```diff
@@ -28,6 +28,10 @@ beforeEach(() => {
             <form string="To-do" class="o_todo_form_view" js_class="todo_form">
                 <field name="name"/>
                 <field name="state" invisible="1"/>
+                <field name="tag_ids" invisible="1"/>
+                <field name="user_ids" invisible="1"/>
+                <field name="priority" invisible="1"/>
+                <field name="personal_stage_type_id" invisible="1"/>
```
Comment (ppr-odoo) [inline addons/project_todo/static/tests/todo_done_checkmark.test.js:34]: same here

File: addons/project_todo/static/tests/todo_conversion_form_view.test.js [hunk 7]
```diff
@@ -15,6 +15,10 @@ beforeEach(() => {
         form: `
             <form string="To-do" class="o_todo_form_view" js_class="todo_form">
                 <field name="name"/>
+                <field name="tag_ids" invisible="1"/>
+                <field name="user_ids" invisible="1"/>
+                <field name="priority" invisible="1"/>
+                <field name="personal_stage_type_id" invisible="1"/>
             </form>`,
```
Comment (ppr-odoo) [inline addons/project_todo/static/tests/todo_conversion_form_view.test.js:22]: same here

File: addons/project_todo/static/src/scss/todo.scss [hunk 8]
```diff
@@ -66,3 +66,33 @@
...
+    }
+    li.nav-item {
+        display: none;
+    }
+}
+
+.o_form_view {
+    .o_form_sheet_bg {
+        max-width: none;
+    }
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/scss/todo.scss:97]:
```suggestion
.o_todo... > .o_form_view {
    .o_form_sheet_bg {
        max-width: none;
    }
```

File: addons/project_todo/static/src/scss/todo.scss [hunk 9]
```diff
@@ -66,3 +66,33 @@
...
+}
+
+.o_control_panel_main_buttons{
+    @include media-breakpoint-up(sm) {
+        margin-inline: 10px;
+    }
+    @include media-breakpoint-down(sm) {
+        margin-right: 6px;
+    }
+}
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/scss/todo.scss:83]: you can use to_do app class

File: addons/project_todo/static/src/scss/todo.scss [hunk 10]
```diff
@@ -66,3 +66,33 @@
 .o_todo_hide_avatar .o_avatar {
     display: none;
 }
+
+.o_tags_input.o_input {
+    @include media-breakpoint-up(md) {
+        overflow: auto;
+    }
+}
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/scss/todo.scss:74]: same here

File: addons/project_todo/static/src/components/todo_user_ids/todo_user_ids.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    Many2ManyTagsAvatarFieldPopover,
+    Many2ManyTagsAvatarField,
+} from "@web/views/fields/many2many_tags_avatar/many2many_tags_avatar_field";
+import { _t } from "@web/core/l10n/translation";
+
+export class TodoUserIds extends Many2ManyTagsAvatarField {
+static template = "todo.TodoMany2ManyTagsAvatarUserField";
+    setup() {
+        super.setup();
+        this.placeholder = _t("Asssignees");
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/components/todo_user_ids/todo_user_ids.js:14]:
```suggestion
        this.placeholder = _t("Asssignees");
```
directly add in the input/field.

File: addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.xml [hunk 12]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<templates xml:space="preserve">
+
+    <t t-name="project_todo.TodoChatter" t-ref="root">
+        <chatter t-if="this.state.displayChatter" class="o_FormRenderer_chatterContainer o_scroll_view_lg">
+            <Chatter
+                threadModel="'project.task'"
+                threadId="this.props.record.resId"
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.xml:9]:
```suggestion
                threadId="record.resId"
```

File: addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+    setup() {
+        this.state = useState({
+            displayChatter: false,
+        });
+    }
+
+    toggleChatter(event) {
+        if (this.props.record.resId) {
+            this.state.displayChatter = !this.state.displayChatter;
+            this.env.bus.trigger('TODO:TOGGLE_CHATTER', {displayChatter: this.state.displayChatter});
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js:56]:
```suggestion
            this.env.bus.trigger('TODO:TOGGLE_CHATTER', { displayChatter: this.state.displayChatter });
```

File: addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+        this.env.bus.addEventListener('TODO:TOGGLE_CHATTER', this.toggleChatter.bind(this));
+    }
+
+    toggleChatter(event) {
+        this.state.displayChatter = event.detail.displayChatter;
+        if (this.state.displayChatter) {
+            this.root.el?.parentElement?.classList.remove('d-none');
+        } else {
+            this.root.el?.parentElement?.classList.add('d-none');
+        }
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js:29]: use toggleClass

File: addons/project_todo/static/src/views/todo_form/todo_form_control_panel.xml [hunk 15]
```diff
@@ -8,14 +8,49 @@
...
+                <t t-slot="todo-tag-ids"/>
+                <t t-slot="todo-user-ids"/>
+                <t t-slot="todo-personal-stage"/>
+            </t>
+            <xpath expr="//div[hasclass('o_control_panel_main_buttons')]" position="move"/>
+            <t t-if="env.isSmall">
+                <t t-slot="todo-personal-stage"/>
+                <t t-slot="todo-user-ids"/>
+                <t t-slot="todo-tag-ids"/>
+            </t>
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/views/todo_form/todo_form_control_panel.xml:22]: We should manage with the order instead.[order](https://getbootstrap.com/docs/5.2/utilities/flex/#order)

File: addons/project_todo/static/src/scss/todo.scss [hunk 16]
```diff
@@ -66,3 +80,30 @@
...
+.o_todo_control_panel, .o_todo_panel_field {
+    @include media-breakpoint-up(md) {
+        overflow: auto;
+    }
+}
+
+.o_todo_chatter {
+    @include media-breakpoint-up(md) {
+        max-width: 30%;
+    }
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/scss/todo.scss:108]: Is it not feasible to achieve this using Bootstrap classes because they are precompiled?

File: addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.xml [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+                hasParentReloadOnFollowersUpdate="true"
+                hasParentReloadOnMessagePosted="true"
+                saveRecord="props.record.save.bind(props.record, { reload: false })"
+            />
+        </chatter>
+    </t>
+
+    <t t-name="project_todo.TodoChatterPanel" t-ref="root">
+        <a role="button"
+            t-attf-class="btn btn-light btn-chatter todo_toggle_chatter #{state.displayChatter ? 'active' : '' }  #{props.record.resId ? '': 'disabled'}" data-hotkey="d" t-on-click="toggleChatter" title="Toggle chatter">
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.xml:21]:
```suggestion
            t-attf-class="btn btn-light btn-chatter todo_toggle_chatter #{state.displayChatter ? 'active' : '' }  #{props.record.resId ? '' : 'disabled'}"
             data-hotkey="d" t-on-click="toggleChatter" title="Toggle chatter">
```

File: addons/project_todo/static/src/views/todo_form/todo_form_arch_parser.js [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+export class TodoArchParser extends FormArchParser {
+    parse(xmlDoc, models, modelName) {
+        let result = super.parse(xmlDoc, models, modelName);
+        let fieldDependencies = {
+            'name_0': getPropertyFieldInfo({ name: 'name', type: 'char' }),
+            'priority_0': getPropertyFieldInfo({ name: 'priority', type: 'selection' }),
+            'tag_ids_0': todogetPropertyFieldInfo({
+                name: 'tag_ids',
+                type: 'many2many',
+                options:"{'color_field': 'color', 'no_create_edit': True}",
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/views/todo_form/todo_form_arch_parser.js:36]: We should pass the value through an object rather than using a string.

File: addons/project_todo/static/src/views/todo_form/todo_form_arch_parser.js [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+                relation: "project.tags",
+            }),
+            'user_ids_0': todogetPropertyFieldInfo({ name: 'user_ids', type: 'many2many', relation: "res.users" }),
+            'personal_stage_type_0': todogetPropertyFieldInfo({
+                name: 'personal_stage_type_id',
+                type: 'many2one',
+                domain: "[('user_id','=',uid)]",
+                options: "{'clickable': '1', 'fold_field': 'fold'}"
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/views/todo_form/todo_form_arch_parser.js:44]: same here

File: addons/project_todo/static/src/views/todo_form/todo_form_arch_parser.js [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+                name: 'tag_ids',
+                type: 'many2many',
+                options:"{'color_field': 'color', 'no_create_edit': True}",
+                relation: "project.tags",
+            }),
+            'user_ids_0': todogetPropertyFieldInfo({ name: 'user_ids', type: 'many2many', relation: "res.users" }),
+            'personal_stage_type_0': todogetPropertyFieldInfo({
+                name: 'personal_stage_type_id',
+                type: 'many2one',
+                domain: "[('user_id','=',uid)]",
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/views/todo_form/todo_form_arch_parser.js:43]:
```suggestion
                domain: "[('user_id', '=' , uid)]",
```

File: addons/project_todo/static/src/components/todo_many2many_users_tags/todo_many2many_users_tags.js [hunk 21]
```diff
@@ -0,0 +1,57 @@
+import { usePopover } from "@web/core/popover/popover_hook";
+import { patch } from "@web/core/utils/patch";
+import {
+    Many2ManyTagsAvatarFieldPopover,
+    Many2ManyTagsAvatarField,
+} from "@web/views/fields/many2many_tags_avatar/many2many_tags_avatar_field";
+import { Many2ManyTagsField } from "@web/views/fields/many2many_tags/many2many_tags_field";
+import { Many2XAutocomplete } from "@web/views/fields/relational_utils";
+import { _t } from "@web/core/l10n/translation";
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/components/todo_many2many_users_tags/todo_many2many_users_tags.js:9]: We should move this patch after because it will maintain grouping (core).

**Comment by ppr-odoo** [comment]
thanks for the work @desh-odoo 

I left some comments and I will check the rest tomorrow.

File: addons/project_todo/static/src/scss/todo.scss [hunk 22]
```diff
@@ -66,3 +79,21 @@
...
+
+@media screen and (min-width: 1488px) {
+    .o_todo_chatter {
+        max-width: 30%;
+    }
+}
+
+.o_todo_m2m_user_tag {
+    .o_delete {
+        color: #5f636f !important;
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/scss/todo.scss:97]:
```suggestion
        color: $o-gray-600;
```

**Review by pkyriakou — COMMENTED** [review]
Thank you @desh-odoo :100: I left some comments for you :grin:

File: addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+    additionalClasses: [
+        'o_todo_chatter',
+        'position-relative',
+        'p-0',
+    ]
+}
+
+registry.category("view_widgets").add("todo_chatter_panel", todoChatter);
+
+export class TodoChatterPanel extends TodoChatter {
```
[the remark this answers is not in the document]
Comment (desh-odoo) [reply addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js:36]: Yes, TodoChatterPanel is only for the button.
Changes Done.

File: addons/project_todo/static/src/views/todo_form/todo_form_control_panel.xml
[the remark this answers is not in the document]
Comment (desh-odoo) [reply addons/project_todo/static/src/views/todo_form/todo_form_control_panel.xml:1]: Changes done

File: addons/project_todo/static/src/views/todo_form/todo_form_control_panel.xml [hunk 24]
```diff
@@ -8,13 +8,55 @@
...
 
+        <xpath expr="//div[hasclass('o_control_panel_navigation')]/t[@t-if='state.embeddedInfos.embeddedActions?.length and env.isSmall']" position="before">
+            <div class="d-flex flex-nowrap align-items-center o_todo_control_panel"
+                t-attf-class="{{ !env.isSmall ? 'row' : '' }}">
+                <div class="order-md-1 order-3 o_todo_panel_field col-md-5"><t t-slot="todo-tag-ids"/></div>
+                <div class="order-md-2 order-2 o_todo_panel_field col-md-5"><t t-slot="todo-user-ids"/></div>
+                <div class="order-md-3 order-1 col-md-2 d-md-flex justify-content-md-end"><t t-slot="todo-personal-stage"/></div>
+            </div>
+            <xpath expr="//div[hasclass('o_control_panel_main_buttons')]" position="move"/>
+            <t t-slot="control-panel-priority-field" />
```
[the remark this answers is not in the document]
Comment (desh-odoo) [reply addons/project_todo/static/src/views/todo_form/todo_form_control_panel.xml:19]: Changes done.

File: addons/project_todo/static/src/components/todo_many2many_users_tags/todo_many2many_users_tags.js
[the remark this answers is not in the document]
Comment (desh-odoo) [reply addons/project_todo/static/src/components/todo_many2many_users_tags/todo_many2many_users_tags.js:1]: Here we managed both tags in one because we have to do similar things.

File: addons/project_todo/static/tests/todo_conversion_form_view.test.js [hunk 25]
```diff
@@ -56,11 +64,15 @@ test("Check that todo_conversion_form view focuses on the focus on the first ele
         };
     });
 
-    await mountView({
-        resModel: "project.task",
+    await mountWithCleanup(WebClient);
+    await getService("action").doAction({
```
[the remark this answers is not in the document]
Comment (desh-odoo) [reply addons/project_todo/static/tests/todo_conversion_form_view.test.js:68]: I think there is no need for a mountWithCleanup we can directly do it with mount view by giving id to the form view. Previously I Didn't know about that that's why I used mountWithCleanup.

File: addons/project_todo/static/tests/todo_conversion_form_view.test.js [hunk 26]
```diff
@@ -11,7 +11,7 @@ defineTodoModels();
 
 beforeEach(() => {
     ProjectTask._views = {
-        form: `
+        "form,1": `
```
Comment (pkyriakou) [inline addons/project_todo/static/tests/todo_conversion_form_view.test.js:14]:
```suggestion
        "form, 1": `
```

File: addons/project_todo/static/tests/todo_form_view.test.js [hunk 27]
```diff
@@ -23,7 +23,7 @@ beforeEach(() => {
                 <field name="name" nolabel="1"/>
                 <field name="state" widget="todo_done_checkmark" nolabel="1"/>
             </tree>`,
-        form: `
+        "form,1": `
```
Comment (pkyriakou) [inline addons/project_todo/static/tests/todo_form_view.test.js:26]:
```suggestion
        "form, 1": `
```

File: addons/project_todo/static/tests/todo_conversion_form_view.test.js [hunk 28]
```diff
@@ -62,6 +62,7 @@ test("Check that todo_conversion_form view focuses on the focus on the first ele
         resId: 1,
         type: "form",
         actionMenus: {},
+        viewId: 1,
```
Comment (pkyriakou) [inline addons/project_todo/static/tests/todo_conversion_form_view.test.js:65]: is that needed? :thinking: shouldn't specifying "form" in the type suffice? 
the same applies to the other test ``todo_form_view.test.js``

File: addons/project_todo/static/tests/tours/project_todo_main_functions.js [hunk 29]
```diff
@@ -88,6 +88,11 @@ registry.category("web_tour.tours").add('project_todo_main_functions', {
 {
     trigger: ".o_todo_form_view",
 },
+{
+    trigger: ".todo_toggle_chatter",
+    content: "Click on chatter button to send message",
```
Comment (pkyriakou) [inline addons/project_todo/static/tests/tours/project_todo_main_functions.js:93]:
```suggestion
    content: "Clicking on the chatter button should toggle open the chatter",
```

**Review by pkyriakou — COMMENTED** [review]
Thank you @desh-odoo for the updates :100: Left a few minor comments :grin:

File: addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js [hunk 30]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+export const todoChatter = {
+    component: TodoChatter,
+    additionalClasses: [
+        'o_todo_chatter',
+        'position-relative',
+        'p-0',
+    ]
+}
```
[the remark this answers is not in the document]
Comment (desh-odoo) [reply addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js:32]: Done

File: addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js [hunk 31]
```diff
@@ -0,0 +1,34 @@
...
+    }
+}
+
+export const todoChatter = {
+    component: TodoChatter,
+    additionalClasses: [
+        'o_todo_chatter',
+        'position-relative',
+        'p-0',
+    ]
```
[the remark this answers is not in the document]
Comment (desh-odoo) [reply addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js:31]: done

File: addons/project_todo/static/src/views/todo_form/todo_form_controller.js [hunk 32]
```diff
@@ -1,12 +1,13 @@
...
 
 import { _t } from "@web/core/l10n/translation";
 import { user } from "@web/core/user";
 import { FormController } from "@web/views/form/form_controller";
-import { TodoDoneCheckmark } from "@project_todo/components/todo_done_checkmark/todo_done_checkmark";
 import { PriorityField } from "@web/views/fields/priority/priority_field";
+import { StatusBarField } from "@web/views/fields/statusbar/statusbar_field";
 
-import { onWillStart } from "@odoo/owl";
+import { TodoDoneCheckmark } from "@project_todo/components/todo_done_checkmark/todo_done_checkmark";
```
[the remark this answers is not in the document]
Comment (desh-odoo) [reply addons/project_todo/static/src/views/todo_form/todo_form_controller.js:9]: It was done, to maintain the order.

File: addons/project_todo/static/tests/todo_conversion_form_view.test.js (hunk 28 above)
Comment (desh-odoo) [reply addons/project_todo/static/tests/todo_conversion_form_view.test.js:65]: Here we need to open the particular view that's why we needed viewId.

File: addons/project_todo/static/tests/tours/project_todo_main_functions.js (hunk 29 above)
Comment (desh-odoo) [reply addons/project_todo/static/tests/tours/project_todo_main_functions.js:93]: Done

File: addons/project_todo/static/tests/todo_conversion_form_view.test.js (hunk 26 above)
Comment (desh-odoo) [reply addons/project_todo/static/tests/todo_conversion_form_view.test.js:14]: By applying space, it cannot find the view because it considers the space also. So, it does not need to change.

File: addons/project_todo/static/tests/todo_form_view.test.js (hunk 27 above)
Comment (desh-odoo) [reply addons/project_todo/static/tests/todo_form_view.test.js:26]: By applying space, it cannot find the view because it considers the space also. So, it does not need to change.

File: addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js [hunk 33]
```diff
@@ -0,0 +1,34 @@
...
+import { registry } from "@web/core/registry";
+import { standardWidgetProps } from '@web/views/widgets/standard_widget_props';
+import { Chatter } from '@mail/chatter/web_portal/chatter';
+
+export class TodoChatter extends Component {
+    static template = "project_todo.TodoChatter";
+    static components = { Chatter };
+    static props = {
+        ...standardWidgetProps,
+    };
```
[the remark this answers is not in the document]
Comment (desh-odoo) [reply addons/project_todo/static/src/components/todo_chatter_panel/todo_chatter_panel.js:11]: Done

**Review by pkyriakou — COMMENTED** [review]
Okay @desh-odoo thanks for your input, I don't have any other comments, LGTM :green_circle:

File: addons/project_todo/views/project_task_views.xml [hunk 34]
```diff
@@ -102,41 +102,45 @@
...
-                    </div>
                     <field name="personal_stage_type_id"
                            domain="[('user_id','=',uid)]"
                            widget="statusbar"
                            options="{'clickable': '1', 'fold_field': 'fold'}"/>
                     <field name="active" invisible="1"/>
-                    <field name="state" invisible="1"/>
                 </header>
-                <sheet class="o_todo_form_sheet_bg">
+                <sheet class="o_todo_form_sheet_bg mw-100">
```
Comment (ppr-odoo) [inline addons/project_todo/views/project_task_views.xml:112]:
```suggestion
                <sheet class="o_todo_form_sheet_bg">
```

File: addons/project_todo/static/src/scss/todo.scss [hunk 35]
```diff
@@ -66,3 +63,9 @@
 .o_todo_hide_avatar .o_avatar {
     display: none;
 }
+
+@media screen and (min-width: 1488px) {
+    .o_todo_chatter {
+        max-width: 30%;
+    }
+}
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/scss/todo.scss:71]:
```suggestion
```

File: addons/project_todo/static/src/views/todo_form/todo_form_control_panel.xml [hunk 36]
```diff
@@ -2,19 +2,12 @@
...
-        </xpath>
-
-    </t>
-
-    <t t-name="project_todo.TodoBreadcrumbs" t-inherit="web.Breadcrumbs" t-inherit-mode="primary">
-        <xpath expr="//div[hasclass('o_last_breadcrumb_item')]//span" position="replace">
-            <t t-slot="control-panel-priority-field" />
-            <t t-slot="control-panel-todo-name" />
-            <t t-slot="todo-mark-as-done-button" />
+        <xpath expr="//div[hasclass('o_control_panel_navigation')]/t[@t-if='state.embeddedInfos.embeddedActions?.length and env.isSmall']" position="before">
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/views/todo_form/todo_form_control_panel.xml:5]:
```suggestion
        <xpath expr="//div[hasclass('o_cp_pager')]" position="before">
```

File: addons/project_todo/static/src/scss/todo.scss [hunk 37]
```diff
@@ -58,6 +55,12 @@
     .oe_description {
         overflow-x: hidden;
     }
+
+    .o_task_state_widget .o_status{
+        width: $font-size-base * 1.36;
+        height: $font-size-base * 1.36;
+        margin-top: -0.5px;
```
Comment (ppr-odoo) [inline addons/project_todo/static/src/scss/todo.scss:62]: code duplication is occurring. [link](https://github.com/odoo/odoo/blob/master/addons/project/static/src/scss/project_widgets.scss#L67)

File: addons/project_todo/views/project_task_views.xml [hunk 38]
```diff
@@ -81,41 +81,45 @@
...
+                                widget="many2many_tags"
+                                options="{'color_field': 'color', 'no_create_edit': True}"
+                                class="me-2"/>
+                        </group>
+                    </group>
+                    <field name="description" type="html" class="oe_description" default_focus="1" options="{'resizable': false, 'collaborative': true}"
+                        placeholder="Type Here..."/>
                 </sheet>
-                <chatter/>
+                <widget name="todo_chatter_panel" groups="base.group_user"/>
```
Comment (xavierbol) [inline addons/project_todo/views/project_task_views.xml:122]: the `groups` attribute does not seem needed since that view is only available for internal users, do I miss something? 🧐

Comment (desh-odoo) [reply addons/project_todo/views/project_task_views.xml:122]: Ok, I removed the attribute.

File: addons/project_todo/views/project_task_views.xml [hunk 39]
```diff
@@ -81,41 +81,45 @@
...
                     <widget name="web_ribbon" class="todo_archived" title="Archived" bg_color="text-bg-danger" invisible="active"/>
-                    <field name="description" type="html" class="oe_description" default_focus="1" options="{'resizable': false, 'collaborative': true}"/>
+                    <div class="oe_title pe-0">
+                        <h1 class="d-flex justify-content-between align-items-center">
+                            <div class="d-flex w-100">
+                                <field name="priority" widget="priority_switch" class="me-3"/>
+                                <field name="name" options="{'line_breaks': False}" widget="text" class="o_task_name text-truncate w-md-75 w-100 pe-2" placeholder="To-do..."/>
+                            </div>
+                            <div class="d-flex justify-content-end o_state_container">
+                                <field name="state" widget="todo_done_checkmark" class="o_task_state_widget" />
```
Comment (pkyriakou) [inline addons/project_todo/views/project_task_views.xml:100]:
```suggestion
                                <field name="state" widget="todo_done_checkmark" class="o_task_state_widget"/>
```

**Review by pkyriakou — COMMENTED** [review]
Thank you @desh-odoo, I left some comments / questions for you :grin:

File: addons/project_todo/views/project_task_views.xml (hunk 39 above)
Comment (desh-odoo) [reply addons/project_todo/views/project_task_views.xml:100]: Changes done

File: addons/project_todo/static/src/views/todo_form/todo_form_control_panel.js [hunk 40]
```diff
@@ -4,4 +4,9 @@ import { ControlPanel } from "@web/search/control_panel/control_panel";
 
 export class TodoFormControlPanel extends ControlPanel {
     static template = "project_todo.TodoFormControlPanel";
+
+    toggleChatter() {
+        this.state.displayChatter = !this.state.displayChatter;
```
[the remark this answers is not in the document]
Comment (desh-odoo) [reply addons/project_todo/static/src/views/todo_form/todo_form_control_panel.js:9]: yes, it is needed because we use it to display chatter icon is active or not.

File: addons/project_todo/static/src/components/todo_chatter/todo_chatter.js [hunk 41]
```diff
@@ -0,0 +1,21 @@
...
+                scrollTo(document.querySelector('.o_FormRenderer_chatterContainer'), { behavior: "smooth" })
+            }
+        });
+    }
+
+    _onMounted() {
+        super._onMounted();
+        this.isChatterScroll = true;
+    }
+}
```
Comment (xavierbol) [inline addons/project_todo/static/src/components/todo_chatter/todo_chatter.js:21]: Actually, what is the purpose of these changes?

Comment (xavierbol) [reply addons/project_todo/static/src/components/todo_chatter/todo_chatter.js:21]: Ok I see, then I think we could do it in TodoPanel instead of creating an override just for that. I will do it.

Comment (ppr-odoo) [reply addons/project_todo/static/src/components/todo_chatter/todo_chatter.js]: Okay, I checked the code and noticed that the chatter always displays on small devices. However, the testing point was to enable scrolling down in the small view, so we had to extend the chatter. thanks

**Review by xavierbol — APPROVED** [review]
robodoo r+
